apps/examples/testcase/ta_tc/eventloop/utc/utc_eventloop_main.c

```diff
@@ -0,0 +1,211 @@
+ *
+ ****************************************************************************/
+
+#include <string.h>
+#include <sys/types.h>
+#include <eventloop/eventloop.h>
+
+#include "tc_common.h"
+
+#define EL_TIMER_DATA          "timer"
+#define EL_TIMER_REPEAT_DATA   "repeat_timer"
+
+static int el_timer_flag;
+static el_timer_t *timer;
+static el_timer_t *timer_repeat;
+
+static void el_timer_cb(void *data)
+{
+	printf("Timer callback is called!\n");
+	if (strncmp((char *)data, EL_TIMER_DATA, strlen(EL_TIMER_DATA) + 1) == 0) {
+		el_timer_flag++;
+		(void)eventloop_delete_timer(timer);
+	}
+}
+
+static void el_timer_repeat_cb(void *data)
+{
+	printf("Timer repeat callback is called!\n");
+	if (strncmp((char *)data, EL_TIMER_REPEAT_DATA, strlen(EL_TIMER_REPEAT_DATA) + 1) == 0) {
+		el_timer_flag++;
+		if (el_timer_flag == 3)	{
+			(void)eventloop_delete_timer(timer_repeat);
+		}
+	}
+}
+
+static void el_delete_timer_cb(void *data)
+{
+	printf("Delete timer callback is called!\n");
+}
+
+static void utc_eventloop_loop_run_n(void)
+{
+}
+
+static void utc_eventloop_loop_run_p(void)
+{
+	int ret;
+
+	ret = eventloop_loop_run();
+	TC_ASSERT_EQ("eventloop_loop_run", ret, OK);
+
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_eventloop_loop_stop_n(void)
+{
+}
+
+static void utc_eventloop_loop_stop_p(void)
+{
+	int ret;
+
+	ret = eventloop_loop_stop();
+	TC_ASSERT_EQ("eventloop_loop_stop", ret, OK);
+
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_eventloop_add_timer_n(void)
+{
+	timer = eventloop_add_timer(1000, 0, NULL, NULL);
+	(void)eventloop_loop_run();
+	TC_ASSERT_EQ("eventloop_add_timer", timer, NULL);
+
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_eventloop_add_timer_p(void)
+{
+	char *timer_data;
+	char *timer_repeat_data;
+
+	timer_data = malloc(sizeof(strlen(EL_TIMER_DATA)));
+	timer_repeat_data = malloc(sizeof(strlen(EL_TIMER_REPEAT_DATA)));
+	strncpy(timer_data, EL_TIMER_DATA, strlen(EL_TIMER_DATA) + 1);
+	strncpy(timer_repeat_data, EL_TIMER_REPEAT_DATA, strlen(EL_TIMER_REPEAT_DATA) + 1);
+	
+	el_timer_flag = 0;
+	timer = eventloop_add_timer(0, 0, el_timer_cb, timer_data);
+	(void)eventloop_loop_run();
+	TC_ASSERT_EQ_CLEANUP("eventloop_add_timer", el_timer_flag, 1, free(timer_data));
+
+	el_timer_flag = 0;
+	timer_repeat = eventloop_add_timer(1000, 1, el_timer_repeat_cb, timer_repeat_data);
+	(void)eventloop_loop_run();
+	TC_ASSERT_EQ_CLEANUP("eventloop_add_timer", el_timer_flag, 3, free(timer_repeat_data));
+
+	free(timer_data);
+	free(timer_repeat_data);
+
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_eventloop_delete_timer_n(void)
+{
+	int ret;
+
+	ret = eventloop_delete_timer(NULL);
+	TC_ASSERT_EQ("eventloop_delete_timer", ret, EVENTLOOP_INVALID_PARAM);
+
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_eventloop_delete_timer_p(void)
+{
+	el_timer_t *del_timer;
+	int ret;
+	
+	el_timer_flag = 0;
+	del_timer = eventloop_add_timer_async(0, 0, el_delete_timer_cb, NULL);
+	ret = eventloop_delete_timer(del_timer);
```

Comment:
I think you'd better check del_timer is okay or NULL before passing to eventloop_delete_timer().